packages/store/addon/-private/system/schema-definition-service.ts

```diff
@@ -103,7 +104,7 @@ export function getModelFactory(store: CoreStore, cache, normalizedModelName: st
     if (klass.isModel) {
       let hasOwnModelNameSet = klass.modelName && Object.prototype.hasOwnProperty.call(klass, 'modelName');
       if (!hasOwnModelNameSet) {
-        klass.modelName = normalizedModelName;
+        Object.defineProperty(klass, 'modelName', { value: normalizedModelName });
```

Comment:
You can probably do this in DEBUG only, if it matters

Reply:
I suppose it'll happen in DEBUG only anyways

Reply:
I considered that but I feel the little bit of extra immutability safety here is nice.